src/Db/Schema/VersionAbstract.php

```diff
@@ -7,32 +7,68 @@
 use Neighborhoods\Kojo\Doctrine;
 use Neighborhoods\Kojo\Doctrine\Connection\DecoratorInterface;
 use Neighborhoods\Pylon\Data\Property\Defensive;
+use Doctrine\DBAL\Types\Type;
 
 abstract class VersionAbstract implements VersionInterface
 {
     use Defensive\AwareTrait;
     use Doctrine\Connection\Decorator\Repository\AwareTrait;
     protected $_createTable;
     protected $_tableName;
+    protected $_mostRecentUnsupportedType;
 
     public function applySchemaSetupChanges(): VersionInterface
     {
         $connection = $this->_getDoctrineConnectionDecoratorRepository()->getConnection(DecoratorInterface::ID_SCHEMA);
         if (!$connection->getSchemaManager()->tablesExist([$this->_getTableName()])) {
-            $connection->beginTransaction();
             try {
-                $this->_assembleSchemaChanges();
-                $connection->getSchemaManager()->createTable($this->_getCreateTable());
-                $connection->commit();
+                $connection->beginTransaction();
+                while (!$this->_canAssembleSchemaChanges()) {
```

Comment:
@p-mucha - this is the one that we should talk through. testing and asking should be two things. we just need to find an elegant approach here. it's close though